plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/standard/AppEngineStandardWizardPage.java

```diff
@@ -35,4 +52,36 @@ public void setHelp(Composite container) {
         "com.google.cloud.tools.eclipse.appengine.newproject.NewStandardProjectContext"); //$NON-NLS-1$
   }
 
+
+  @Override
+  protected void createRuntimeField(Composite composite) {
+    Label runtimeLabel = new Label(composite, SWT.LEAD);
+    runtimeLabel.setText(Messages.getString("runtime")); //$NON-NLS-1$
+    runtimeField = new ComboViewer(composite, SWT.READ_ONLY);
+    runtimeField.setLabelProvider(new LabelProvider() {
+      @Override
+      public String getText(Object element) {
+        return ((AppEngineRuntime) element).getLabel();
+      }
+    });
+    runtimeField.setContentProvider(ArrayContentProvider.getInstance());
+    runtimeField.setInput(AppEngineRuntime.STANDARD_RUNTIMES);
+    runtimeField.setSelection(new StructuredSelection(DEFAULT_RUNTIME), true);
+    runtimeField.addPostSelectionChangedListener(new ISelectionChangedListener() {
+      @Override
+      public void selectionChanged(SelectionChangedEvent event) {
+        revalidate();
+      }
+    });
+  }
+
+  @Override
+  public String getRuntimeId() {
+    ISelection selection = runtimeField == null ? new StructuredSelection(DEFAULT_RUNTIME)
```

Comment:
This can be `ISelection selection = runtimeField == null ? null : runtimeField.getSelection();` because of the check below. (`null instanceof ...` works.)

But better yet, I'd just do

```java
if (runtimeField == null) {
  return DEFAULT_RUNTIME.getId();
}
```

But again, is it possible that `runtimeField` is null? This class is the standard wizard page that always create the field.